packages/fx-core/src/question/create.ts

```diff
@@ -1091,18 +1093,26 @@ export function apiSpecLocationQuestion(includeExistingAPIs = true): SingleFileO
   };
 }
 
-export function apiMessageExtensionAuthQuestion(): SingleSelectQuestion {
+export function apiAuthQuestion(): SingleSelectQuestion {
   return {
     type: "singleSelect",
-    name: QuestionNames.ApiMEAuth,
+    name: QuestionNames.ApiAuth,
     title: getLocalizedString("core.createProjectQuestion.apiMessageExtensionAuth.title"),
     placeholder: getLocalizedString(
       "core.createProjectQuestion.apiMessageExtensionAuth.placeholder"
     ),
     cliDescription: "The authentication type for the API.",
-    staticOptions: ApiMessageExtensionAuthOptions.all(),
-    dynamicOptions: () => ApiMessageExtensionAuthOptions.all(),
-    default: ApiMessageExtensionAuthOptions.none().id,
+    staticOptions: ApiAuthOptions.all(),
+    dynamicOptions: (inputs: Inputs) => {
+      const options: OptionItem[] = [ApiAuthOptions.none()];
+      if (inputs[QuestionNames.MeArchitectureType] == MeArchitectureOptions.newApi().id) {
+        options.push(ApiAuthOptions.apiKey(), ApiAuthOptions.microsoftEntra());
+      } else if (inputs[QuestionNames.Capabilities] == CapabilityOptions.copilotPluginNewApi().id) {
+        options.push(ApiAuthOptions.apiKey(), ApiAuthOptions.oauth());
```

Comment:
Do we need feature for these two options? We have "isCopilotAuthEnabled" for Copilot plugin with auth from existing API. @SLdragon please correct me if I am wrong.

Reply:
Let me add the `isCopilotAuthEnabled` feature flag in next PR, thanks!